test/EspressoTEEVerifier.t.sol

```diff
@@ -0,0 +1,103 @@
+    address adminTEE = address(141);
+    address fakeAddress = address(145);
+
+    EspressoTEEVerifier espressoTEEVerifier;
+    EspressoSGXTEEVerifier espressoSGXTEEVerifier;
+    bytes32 enclaveHash =
+        bytes32(0x01f7290cb6bbaa427eca3daeb25eecccb87c4b61259b1ae2125182c4d77169c0);
+    bytes32 enclaveSigner =
+        bytes32(0x5fc862cb2e7e1f449f36a18b18aca08c20feaed0d411247816c281d596420cbb);
+    //  Address of the automata V3QuoteVerifier deployed on sepolia
+    address v3QuoteVerifier = address(0x6E64769A13617f528a2135692484B681Ee1a7169);
+
+    function setUp() public {
+        vm.createSelectFork("https://rpc.ankr.com/eth_sepolia");
+        // Get the instance of the DCAP Attestation QuoteVerifier on the Arbitrum Sepolia Rollup
+        vm.startPrank(adminTEE);
+
+        espressoSGXTEEVerifier =
+            new EspressoSGXTEEVerifier(enclaveHash, enclaveSigner, v3QuoteVerifier);
+        espressoTEEVerifier = new EspressoTEEVerifier(espressoSGXTEEVerifier);
+        vm.stopPrank();
+    }
+
+    function testRegisterSigner() public {
+        string memory quotePath = "/test/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+        address batchPosterAddress = address(0xe2148eE53c0755215Df69b2616E552154EdC584f);
+        bytes memory data = abi.encodePacked(batchPosterAddress);
+        espressoTEEVerifier.registerSigner(sampleQuote, data, IEspressoTEEVerifier.TeeType.SGX);
+    }
+
+    function testRegisteredSigners() public {
+        string memory quotePath = "/test/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+        address batchPosterAddress = address(0xe2148eE53c0755215Df69b2616E552154EdC584f);
+        bytes memory data = abi.encodePacked(batchPosterAddress);
+        espressoTEEVerifier.registerSigner(sampleQuote, data, IEspressoTEEVerifier.TeeType.SGX);
+
+        assertEq(
+            espressoTEEVerifier.registeredSigners(
+                batchPosterAddress, IEspressoTEEVerifier.TeeType.SGX
+            ),
+            true
+        );
+    }
+
+    function testRegisteredEnclaveHash() public {
+        assertEq(
+            espressoTEEVerifier.registeredEnclaveHashes(
+                bytes32(0x01f7290cb6bbaa427eca3daeb25eecccb87c4b61259b1ae2125182c4d77169c0),
+                IEspressoTEEVerifier.TeeType.SGX
+            ),
+            true
+        );
+    }
+
+    function testRegisteredEnclaveSigner() public {
+        assertEq(
+            espressoTEEVerifier.registeredEnclaveSigners(
+                bytes32(0x5fc862cb2e7e1f449f36a18b18aca08c20feaed0d411247816c281d596420cbb),
+                IEspressoTEEVerifier.TeeType.SGX
+            ),
+            true
+        );
+    }
+
+    function testSetEspressoSGXTEEVerifier() public {
```

Comment:
I think similarly here we could encode the test to show that non admins cannot set the address for the sgx verifier. I think that would alert future readers that changes that allow those that aren't the admin to set the sgx verifier address is an undesirable characteristic for the contract.

Reply:
https://github.com/EspressoSystems/espresso-tee-contracts/pull/2/commits/a2bf9e83c0441bd275e4cb10f5bb23b8d440a8ad